### PR TITLE
FEAT: Crafty crate added!

### PR DESCRIPTION
**Pull request by chipsams**
also of note: open crate now accessible in furnished device, alternate pre-wool recipe for furnished device.

### CODE REVIEW

File: kubejs/server_scripts/base/featrues/devices.js [hunk 1]
```diff
@@ -77,14 +77,33 @@ if (
...
+      included_devices: ['botania:crafty_crate','minecraft:oak_door', 'minecraft:oak_sign', 'minecraft:light_blue_bed', 'minecraft:composter', 'minecraft:barrel', 'minecraft:campfire', 'create:cuckoo_clock', 'create:wooden_bracket', 'create:white_seat', 'supplementaries:notice_board', 'supplementaries:speaker_block', 'supplementaries:pulley_block', 'supplementaries:bellows', 'supplementaries:hanging_sign_oak', 'storagedrawers:oak_full_drawers_1', 'minecraft:target', 'minecraft:oak_trapdoor', 'minecraft:note_block', 'minecraft:lectern', 'minecraft:jukebox', 'minecraft:chest', 'minecraft:bookshelf', 'supplementaries:sign_post_oak', 'storagedrawers:oak_full_drawers_2', 'minecraft:item_frame', 'storagedrawers:oak_full_drawers_4', 'minecraft:painting'],
       amount_crafted: 4,
       post_logic: () => {
+
+        //needs to be easy to access, so people who know what they're doing can benefit from the crafty crate before getting into botania.
+        ServerEvents.recipes(e=>{
+          e.recipes.create.deploying("botania:placeholder",["#forge:concrete",Ingredient.of(["botania:crafty_crate","create:mechanical_crafter"])]).keepHeldItem();
```
Comment (Yarden-zamir) [inline kubejs/server_scripts/base/featrues/devices.js:98]: Maybe a slot cover makes more sense? As it does the same thing for create autocrafting. Hm.. I wish we could just combine them

File: kubejs/server_scripts/base/featrues/devices.js [hunk 2]
```diff
@@ -77,14 +77,33 @@ if (
...
+        });
+
+        //patterns kinda make some things too easy, e.g. unpacking storage blocks. we decided that packing is fine though, although we could probably change our minds later.
```
Comment (Yarden-zamir) [inline kubejs/server_scripts/base/featrues/devices.js:101]: I think remove patterns is not related to devices, should be it's own feature in a seperate file

Comment (Yarden-zamir) [reply kubejs/server_scripts/base/featrues/devices.js:101]: Also, you can removeItems instead of your own loop

**Review by Yarden-zamir — COMMENTED** [review]
Looks good, should move the pattern removal into it's own feature and perhapse think abit about the relation between slot cover and placeholder

**Review by Yarden-zamir — CHANGES_REQUESTED** [review]
Looks good, should move the pattern removal into it's own feature and perhapse think abit about the relation between slot cover and placeholder

**Comment by chipsams** [comment]
> Looks good, should move the pattern removal into it's own feature and perhapse think abit about the relation between slot cover and placeholder

@Yarden-zamir ok, made them interchangable like we discussed on discord, although we are still yet to finalize on where they come from such that they are available with next to no progression.

File: kubejs/server_scripts/base/featrues/devices.js [hunk 3]
```diff
@@ -77,14 +77,41 @@ if (
...
+      included_devices: ['botania:crafty_crate','minecraft:oak_door', 'minecraft:oak_sign', 'minecraft:light_blue_bed', 'minecraft:composter', 'minecraft:barrel', 'minecraft:campfire', 'create:cuckoo_clock', 'create:wooden_bracket', 'create:white_seat', 'supplementaries:notice_board', 'supplementaries:speaker_block', 'supplementaries:pulley_block', 'supplementaries:bellows', 'supplementaries:hanging_sign_oak', 'storagedrawers:oak_full_drawers_1', 'minecraft:target', 'minecraft:oak_trapdoor', 'minecraft:note_block', 'minecraft:lectern', 'minecraft:jukebox', 'minecraft:chest', 'minecraft:bookshelf', 'supplementaries:sign_post_oak', 'storagedrawers:oak_full_drawers_2', 'minecraft:item_frame', 'storagedrawers:oak_full_drawers_4', 'minecraft:painting'],
       amount_crafted: 4,
       post_logic: () => {
+
+        
+        if(feature("Cheap Covers & Placeholders")){
```
Comment (Yarden-zamir) [inline kubejs/server_scripts/base/featrues/devices.js:97]: Can you move this feature outside though? This is not related to devices.

File: kubejs/server_scripts/base/featrues/devices.js [hunk 4]
```diff
@@ -77,14 +77,41 @@ if (
...
+            //TODO: can't do keepHeldItem() without a recipe event, probably needs fixing in addDeploying() but i can't be bothered to work out how you do that via recipe json.
+            e.recipes.create.deploying("botania:placeholder",["#forge:concrete","botania:crafty_crate"]).keepHeldItem();
+            e.recipes.create.deploying("create:crafter_slot_cover",["#forge:concrete","create:mechanical_crafter"]).keepHeldItem();
+          });
+  
+          addStonecutting("botania:placeholder","create:crafter_slot_cover");
+          addStonecutting("create:crafter_slot_cover","botania:placeholder");
+        }
+
+        if(feature("Remove Crafty Crate Patterns")){
```
Comment (Yarden-zamir) [inline kubejs/server_scripts/base/featrues/devices.js:110]: Same here. Should be in it's own independent feature not nested under devices

**Review by Yarden-zamir — CHANGES_REQUESTED** [review]
Awesome. Can you move those features into their own file though?

**Review by Yarden-zamir — APPROVED** [review]

**Pull request merged**